Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -178,6 +178,20 @@ if (AMReX_HIP)
 
    set(_valid_hip_compilers clang++ hipcc nvcc CC)
    get_filename_component(_this_comp ${CMAKE_CXX_COMPILER} NAME)
+   
+   # work-arounds for breaking Fortran compiles
+   #   https://github.com/ROCm-Developer-Tools/HIP/pull/2280
+   set(IS_PRE_ROCM4_HIPCC FALSE)
+   # CMake<3.21 and hipcc as CXX
+   if(${_this_comp} STREQUAL hipcc)
+      set(IS_PRE_ROCM4_HIPCC TRUE)
+   endif()
+   # CMake >=3.21 and hipcc as CXX but then set by cmake to clang++
+   if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "ROCMClang")
```

Comment:
Becomes vanilla Clang again in 3.21.3+:
https://gitlab.kitware.com/cmake/cmake/-/merge_requests/6533

Reply:
Fixed because this change was reverted after my report.